### ISSUE TITLE
catch keyword is mis-detected

### ISSUE THREAD

**Issue by ptarjan** [jscs-dev/node-jscs#1855]
I can have an es6 class with a method named `catch` (which the `Promise` class has). It shouldn't be detected as a keyword. Right now I'm getting:

```
One space required after "catch" keyword at file.js :
    45 |  }
    46 |
    47 |  catch(err) {
---------------^
    48 |    return this.promise.then(null, err);
    49 |  }
```

For this code:

```
class Promise {
...
  catch(err) {
    return this.promise.then(null, err);
  }
```

**Comment by hzoo** [comment]
Thanks a lot for the report! This is for http://jscs.info/rule/requireSpaceAfterKeywords right? You can use `--verbose` to verify. Let us know if you want to make a PR - otherwise I can take this

**Comment by hzoo** [comment]
since catch will be a keyword when checking by tokens - do we just need to check if the node is a method?

**Comment by ptarjan** [comment]
I probably won't send the PR, sorry. But yes, it is contextual keyword so you have to check the node type to know if it is a keyword. I don't know what parser you use under the covers but on HHVM (the only parser I've written) we would annotate the nodes differently based on context.

**Comment by mikesherov** [comment]
Another casualty of rules that use iterateTokens*

**Comment by markelog** [comment]
@mikesherov yes, but that token shouldn't have a `Keyword` type, but an `Identifier`, i see this issue in Esprima and in babylon too

**Comment by hzoo** [comment]
Will be fixed in 2.3.1
